src/App.js

```diff
@@ -3,28 +3,34 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(
+const App = () => {
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
 
-    JSON.parse(localStorage.getItem('saveTodoList')) || []
-  )
 
   React.useEffect(() => {
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem('savedTodoList')) || []
+          }
+        })
+      }, 2000)
+    })
+      .then((result) => {
+        setTodoList(result.data.todoList);
```

Comment:
This works for now. However, it can lead to unexpected bugs. React expects a new reference within the setter function to trigger a re-render, it is possible that it could see this as the same reference and not trigger the re-render. To be safe I recommend passing your new array as: `setTodoList([...todos]).

Now you guarantee a new reference to your setter function, triggering a re-render.